core/shared/src/main/scala/zio/stm/ZSTM.scala

```diff
@@ -1125,6 +1145,13 @@ object ZSTM {
       tx.zipWith(f(a))(_ += _)
     }.map(_.result())
 
+  /**
+   * Applies the function `f` to each element of the `Set[A]` and returns a
+   * transactional effect that produces a new `Set[B]`.
+   */
+  def foreach[R, E, A, B](in: Set[A])(f: A => ZSTM[R, E, B]): ZSTM[R, E, Set[B]] =
```

Comment:
Should we have the one for `foreach_` as well?

Reply:
We don't need it because that one just takes an `Iterable` since it doesn't need to honor the "returns the same collection type" guarantee.

Reply:
Good point.